src/unix/thread.c

```diff
@@ -875,3 +890,49 @@ void uv_key_set(uv_key_t* key, void* value) {
   if (pthread_setspecific(*key, value))
     abort();
 }
+
+
+int uv__thread_setname(const char* name) {
+#if defined(UV_PTHREAD_MAX_NAMELEN_NP)
+  char namebuf[UV_PTHREAD_MAX_NAMELEN_NP];
+  strncpy(namebuf, name, sizeof(namebuf) - 1);
+  namebuf[sizeof(namebuf) - 1] = '\0';
+#if defined(__APPLE__)
+  int err = pthread_setname_np(namebuf);
```

Comment:
Are you sure about it setting errno? It might, I couldn't tell from looking at its source code, but it's very uncommon for pthreads functions.

Reply:
Yes, at least that's what I see in my `macOS` mini

Reply:
FWIW I'm on macOS 14 (Sonoma) and that function is documented to return `void`.

Reply:
Really? This is how they're defined in the `/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/include/pthread/pthread.h` file:
```
/*SPI to set and get pthread name*/
__API_AVAILABLE(macos(10.6), ios(3.2))
int     pthread_getname_np(pthread_t,char*,size_t);

__API_AVAILABLE(macos(10.6), ios(3.2))
__PTHREAD_SWIFT_UNAVAILABLE_FROM_ASYNC("Thread lifecycle is owned by Swift Concurrency runtime")
int     pthread_setname_np(const char*);
```
which seems to imply they were available from `10.6` ?

Reply:
I suppose the man page is wrong then!

Reply:
If you could test it locally it would help (or at least check the header file in your system)

Reply:
I see it returning int in both the CLT and the Xcode headers, so you did good!